lightning/src/ln/functional_test_utils.rs

```diff
@@ -85,12 +94,28 @@ pub fn connect_block<'a, 'b, 'c, 'd>(node: &'a Node<'b, 'c, 'd>, block: &Block,
 	node.chain_monitor.chain_monitor.block_connected(&block.header, &txdata, height);
 	node.node.block_connected(&block.header, &txdata, height);
 	node.node.test_process_background_events();
+	node.blocks.lock().unwrap().push((block.header, height));
 }
 
 pub fn disconnect_block<'a, 'b, 'c, 'd>(node: &'a Node<'b, 'c, 'd>, header: &BlockHeader, height: u32) {
 	node.chain_monitor.chain_monitor.block_disconnected(header, height);
 	node.node.block_disconnected(header);
-	node.node.test_process_background_events();
+	node.blocks.lock().unwrap().pop();
+}
+pub fn disconnect_blocks<'a, 'b, 'c, 'd>(node: &'a Node<'b, 'c, 'd>, count: u32) {
+	assert!(node.blocks.lock().unwrap().len() as u32 > count); // Cannot disconnect genesis
+	for _ in 0..count {
+		let (block_header, height) = {
+			let blocks = node.blocks.lock().unwrap();
+			(blocks[blocks.len() - 1].0, blocks[blocks.len() - 1].1)
+		};
```

Comment:
Wondering if this could be simplified a bit without the mutex. Is there a need for it?

Reply:
Oops, indeed. We just need interior mutability, so I swapped it for a refcell.